articles/introduction-to-scikit-learn-in-python/index.md

```diff
@@ -0,0 +1,297 @@
+label_encoding = OneHotEncoder()
+
+# Fit the labels
+encoded = label_encoding.fit(band_list)
+
+print(encoded.transform(band_list).toarray())
+
+# Output
+# [[1. 0. 0. 0. 0. 0. 0. 0.]
+#  [0. 1. 0. 0. 0. 0. 0. 0.]
+#  [0. 0. 1. 0. 0. 0. 0. 0.]
+#  [0. 0. 0. 1. 0. 0. 0. 0.]
+#  [0. 0. 0. 0. 1. 0. 0. 0.]
+#  [0. 0. 0. 0. 0. 1. 0. 0.]
+#  [0. 0. 0. 0. 0. 0. 1. 0.]
+#  [0. 0. 0. 0. 0. 0. 0. 1.]]
+```
+
+### Scikit-Learn for Filling Missing Values
+
+Almost 70% of time and resources are spent on collecting and cleaning the dataset for every project. And when one deals with a real-life dataset, there are always missing values. Cleaning the dataset and handling missing data is important as many machine learning algorithms do not accommodate a missing attribute in the data. This is where Scikit-Learn's impute module comes into play. A simple way to deal with missing values is to remove the row of data with a missing value, which would mean losing valuable-yet-incomplete data. A better way is to replace the missing values with values that can be inferred from known data. One way would be to replace the missing data with the mean of that column.
+
+Missing values are encoded with NumPy's NaN (numpy.nan)
+
+The following are the temperatures recorded in Bloomington (in Fahrenheits) in Illinois in the month of February: [33.2,32.8,32.9,33.0,nan,33.2,33.4,33.1,32.6,32.5,32.5,33.1,33.0,nan,32.7,32.7,32.6,nan,32.6,32.9,32.8,32.8,32.5,32.6,nan,32.6,32.7,32.7,33.5, 33.7,33.9].
+
+Let us try to replace the missing temperatures with their mean.
+
+```py
+import numpy as np
+from sklearn.impute import SimpleImputer
+
+#List of temperatures
+temperatures = [33.2,32.8,32.9,33.0,"NaN",33.2,33.4,33.1,32.6,32.5,32.5,33.1,33.0,"NaN",32.7,32.7,32.6,"NaN",32.6,32.9,32.8,
+                32.8,32.5,32.6,"NaN",32.6,32.7,32.7,33.5, 33.7,33.9]
+
+temperatures_cleaned = []
+
+#Replace NaN's with np.nan
+for temperature in temperatures:
+    if temperature=="NaN":
+        temperatures_cleaned.append(np.nan)
+    else:
+        temperatures_cleaned.append(temperature)
+
+temperatures_np = np.array(temperatures_cleaned).reshape(-1,1)
+
+# Create an instance of the imputer
+imputer = SimpleImputer(missing_values=np.nan,strategy='mean')
+
+#Transform the array and fit according to the chosen strategy
+temperatures_np = imputer.fit_transform(temperatures_np)
+
+print(*temperatures_np, sep=", ")
+
+# Output - [33.2], [32.8], [32.9], [33.], [32.91111111], [33.2], [33.4], [33.1], [32.6], [32.5], [32.5], [33.1], [33.], [32.91111111], 
+#          [32.7], [32.7], [32.6], [32.91111111], [32.6], [32.9], [32.8], [32.8], [32.5], [32.6], [32.91111111], [32.6], 
+#          [32.7], [32.7], [33.5], [33.7], [33.9]
+```
+SimpleImputer provides four options for strategy - mean, median, most_frequent, and constant. Since mean was the chosen strategy, the nan's were replaced with the mean of the temperatures (32.91111111). Had most_frequent been the chosen category,
```

Comment:
`SimpleImputer` provides four options for strategy... Consider formatting variables, functions, and anything related to the code while explaining them.

Reply:
Fixed it in the recent commit.